config/routes.rb

```diff
@@ -19,6 +19,7 @@
   resources :nodes, only: [:index, :show, :destroy] do
     collection do
       post :bootstrap
+      post :assign_roles
```

Comment:
This is not very *Resty*. Would it make sense to create a new controllers `RolesController` and then add this functionality as its `update` method ?